src/diffusers/models/attention.py

```diff
@@ -375,16 +448,32 @@ def __init__(self, dim_in: int, dim_out: int):
 
         self.proj = linear_cls(dim_in, dim_out * 2)
 
-    def gelu(self, gate: torch.Tensor) -> torch.Tensor:
-        if gate.device.type != "mps":
-            return F.gelu(gate)
-        # mps: gelu is not implemented for float16
-        return F.gelu(gate.to(dtype=torch.float32)).to(dtype=gate.dtype)
-
     def forward(self, hidden_states, scale: float = 1.0):
         args = () if USE_PEFT_BACKEND else (scale,)
         hidden_states, gate = self.proj(hidden_states, *args).chunk(2, dim=-1)
-        return hidden_states * self.gelu(gate)
+        return hidden_states * gelu(gate)
+
+
+# TODO Name?
+class GatedGelu(nn.Module):
+    def __init__(self, dim_in: int, dim_out: int, bias: bool):
+        super().__init__()
+        linear_cls = LoRACompatibleLinear if not USE_PEFT_BACKEND else nn.Linear
+
+        self.proj1 = linear_cls(dim_in, dim_out, bias)
+        self.proj2 = linear_cls(dim_in, dim_out, bias)
+
+    def forward(self, hidden_states, scale: float = 1.0):
+        args = () if USE_PEFT_BACKEND else (scale,)
+
+        return gelu(self.proj1(hidden_states, *args)) * self.proj2(hidden_states, *args)
```

Comment:
I'm not hip with the variety of gelu variants. Is there a better name for this?

Reply:
What's the reference did you follow for this variant of GeLU?

Reply:
removed!